src/transformers/models/convnext/modeling_flax_convnext.py

```diff
@@ -0,0 +1,627 @@
+
+    def __call__(self, hidden_states: jnp.ndarray) -> jnp.ndarray:
+        hidden_states = self.downsampling_layer(hidden_states)
+        hidden_states = self.layers(hidden_states)
+        return hidden_states
+
+
+class FlaxStageCollection(nn.Module):
+    config: ConvNextConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        # np.split requires list with entries indicating where along axis the array is split
+        sections = (np.cumsum(self.config.depths) - self.config.depths)[1:]
+
+        drop_path_rates = [
+            x.tolist()
+            for x in np.split(
+                np.linspace(0, self.config.drop_path_rate, sum(self.config.depths), dtype=self.dtype),
+                sections,
+            )
+        ]
+
+        prev_channels = [
+            self.config.hidden_sizes[0],
+            *[self.config.hidden_sizes[i] for i in range(self.config.num_stages - 1)],
+        ]
+
+        self.stages = [
+            FlaxConvNextStage(
+                self.config,
+                in_channels=prev_channels[i],
+                out_channels=self.config.hidden_sizes[i],
+                stride=2 if i > 0 else 1,
+                depth=self.config.depths[i],
+                drop_path_rates=drop_path_rates[i],
+                name=str(i),
+                dtype=self.dtype,
+            )
+            for i in range(self.config.num_stages)
+        ]
+
+    def __call__(
+        self,
+        hidden_states: jnp.ndarray,
+        output_hidden_states: Optional[bool] = False,
+        return_dict: Optional[bool] = True,
+    ) -> Tuple[jnp.ndarray, Tuple]:
+        all_hidden_states = () if output_hidden_states else None
+
+        for i, layer_module in enumerate(self.stages):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states.transpose(0, 3, 1, 2),)
+
+            hidden_states = layer_module(hidden_states)
+
+        return hidden_states, all_hidden_states
+
+
+class FlaxConvNextEncoder(nn.Module):
+    config: ConvNextConfig
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.stages = FlaxStageCollection(self.config, self.dtype)
+
+    def __call__(
+        self,
+        hidden_states: jnp.ndarray,
+        output_hidden_states: Optional[bool] = False,
+        return_dict: Optional[bool] = True,
+    ) -> Union[Tuple, FlaxBaseModelOutputWithNoAttention]:
+        hidden_states, all_hidden_states = self.stages(hidden_states, output_hidden_states, return_dict)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states.transpose(0, 3, 1, 2),)
+
+        if not return_dict:
+            return tuple(v for v in [hidden_states, all_hidden_states] if v is not None)
+
+        return FlaxBaseModelOutputWithNoAttention(
+            last_hidden_state=hidden_states,
+            hidden_states=all_hidden_states,
+        )
+
+
+class FlaxConvNextPreTrainedModel(FlaxPreTrainedModel):
```

Comment:
Can this now be `# Copied from ...` one of our Flax ResNet models?

Reply:
No, because we don't have the batch norm layers, we cannot input `batch_stats` here

```python
self.module.apply(
            {"params": params or self.params}
```

Reply:
Thanks for clarifying!